File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListItem.java
package io.jeffchang.buttondemo.ui.userlist.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

import io.jeffchang.buttondemo.R;

/**
 * Created by jeffreychang on 3/4/18.
 */

class UserListItem extends RelativeLayout {

    public UserListItem(Context context) {
        super(context);
        init(context, null);
    }

    public UserListItem(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public UserListItem(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        inflate(context, R.layout.dog_breed_list_item, this);
    }
}

File: app/src/main/java/io/jeffchang/buttondemo/ui/userlist/view/UserListItem.kt
package io.jeffchang.buttondemo.ui.userlist.view

import android.content.Context
import android.widget.RelativeLayout

import io.jeffchang.buttondemo.R
import kotlinx.android.synthetic.main.user_list_item.view.*

/**
 * Custom view that lists user items in a list.
 */

class UserListItem constructor(context: Context)
    : RelativeLayout(context, null, 0) {

    var name: String? = null
        set(fullName) {
            user_list_item_name_text_view.text = fullName
        }

    var email: String? = null
        set(fullName) {
            user_list_item_email_text_view.text = fullName
        }

    init {
        inflate(context, R.layout.user_list_item, this)
    }
}
